test/e2e/framework.go

```diff
@@ -903,21 +913,30 @@ func (data *TestData) deployFlowVisibilityClickHouse(o flowVisibilityTestOptions
 			return true, nil
 		}
 	}); err != nil {
-		return "", fmt.Errorf("timeout waiting for ClickHouse Service: %v", err)
+		return "", fmt.Errorf("timeout waiting for ClickHouse Service: %w", err)
 	}
 
+	const probePodName = "ch-svc-probe"
+	if err := NewPodBuilder(probePodName, flowVisibilityNamespace, agnhostImage).Create(testData); err != nil {
+		return "", fmt.Errorf("failed to create ClickHouse Service probe Pod: %w", err)
+	}
+	defer testData.DeletePod(flowVisibilityNamespace, probePodName)
+	if err := data.podWaitForReady(defaultTimeout, probePodName, flowVisibilityNamespace); err != nil {
+		return "", err
+	}
+
+	cmd := []string{"/agnhost", "connect", net.JoinHostPort(chSvc.Spec.ClusterIP, clickHouseHTTPPort), "--timeout=5s"}
 	if err := wait.PollUntilContextTimeout(context.TODO(), defaultInterval, defaultTimeout, true, func(ctx context.Context) (bool, error) {
-		rc, stdout, stderr, err := testData.RunCommandOnNode(controlPlaneNodeName(),
-			fmt.Sprintf("curl -Ss %s:%s", chSvc.Spec.ClusterIP, clickHouseHTTPPort))
-		if rc != 0 || err != nil {
-			log.Infof("Failed to curl clickhouse Service: %s", strings.Trim(stderr, "\n"))
+		_, stderr, err := testData.RunCommandFromPod(flowVisibilityNamespace, probePodName, agnhostContainerName, cmd)
+		if err != nil {
+			log.Infof("Failed to connnect to clickhouse Service, err: %v, stderr: %s", err, strings.Trim(stderr, "\n"))
 			return false, nil
 		} else {
-			log.Infof("Successfully curl'ed clickhouse Service: %s", strings.Trim(stdout, "\n"))
+			log.Infof("Successfully connected to clickhouse Service")
```

Comment:
Do we use infof or infoS?

Reply:
This is a different logging library (not `klog`) that we use in some parts of the e2e tests (`github.com/sirupsen/logrus`). While logrus does support structured logging, there is no `log.InfoS` function, it works differently. We don't use structured logging in e2e tests.